### ISSUE TITLE
Query string encoding/decoding doesn't work for some characters

### ISSUE THREAD

**Issue by jensli** [shabiel/M-Web-Server#49]
```
GTM>w $$URLDEC^%webutils("%C3%85")
Ã
```

Should produce `Å`.

```
GTM>w $$URLENC^%webutils("Å")
Å
```

Should produce `%C3%85`.

See here for example:

https://www.coderstool.com/querystring-encode
https://stackoverflow.com/a/31300627/452775

### Environment

M Web Server version: 1.1.2
```
GTM>w $zversion
GT.M V7.0-000 Linux x86_64
GTM>w $zchset
UTF-8
```

**Comment by jensli** [comment]
Tip to whomever fixes this: GTM has the following routines for handing the correct encoding:

```
GTM>write $$FUNC^%UTF2HEX("Å")
C385
GTM>write $$FUNC^%HEX2UTF("C385")
Å
```

**Comment by shabiel** [comment]
Jens,

It will be a long time till I get to this as I am traveling. Can you supply
a cross-platform fix and test and I can merge it?

**Comment by jensli** [comment]
I am managed to fix the problem on GT.M.

I'm not 100 % sure how the fix works, thought...

Here is the code:

```
URLDEC(x) ; Decode a URL-encoded string
 ; Q $ZCONVERT(X,"I","URL")  ; uncomment for fastest performance on Cache
 ;
 N I,OUT,FRAG,ASC
 F I=1:1:$L(X,"%") D
 .I I=1 S OUT=$P(X,"%") Q
 .S FRAG=$P(X,"%",I)
 .S ASC=$E(FRAG,1,2)
 .S FRAG=$E(FRAG,3,$L(FRAG))
 .I $L(ASC) S OUT=OUT_$$FUNC^%HEX2UTF(ASC)
 .S OUT=OUT_FRAG
 Q OUT
```

The only change is that I have replaced `$C($$HEX2DEC(ASC))` by `$$FUNC^%HEX2UTF(ASC)`.

I believe that the fix is GTM specific. I have unfortunately no way to find a cross-platform fix since I only have a GTM system. Because of this I don't think I can provide an acceptable patch. I post the GTM fix here in case it is useful to somebody.

**Comment by shabiel** [comment]
@jensli Are you running this in UTF-8 mode or M mode? I looked at this today a little bit as I am working on the source code, and it may related to the mode, and it can easily fixed I think.

**Comment by jensli** [comment]
Yes, we are using UTF-8 mode. It seems reasonable that it is related.

We have also encountered some other problems related to UTF-8:

When the content of the MTTP message body is an octet stream then GTM `$length` and other string routines chokes on non-UTF-8 byte sequences.

The fix for us was the replace `$length` with `$zlength` in the M Web Server source code.

I have the intention to report these problems here when I have some time, and also submit our fixes. Even though I don't think they are portable with Cache maybe they can serve as a starting point.

Apart from a few such bugs M Web Server has been working well for us. Thank you for your work on it!

**Comment by shabiel** [comment]
@jensli I already use `$ZL` in the M Web Server behind a flag for GTM. Please send in a pull request with the changes.

**src/_webrsp.m**:
```
313: I $P($SY,",")=47,$ZL(%WBUFF)+$ZL(DATA)>32000 D FLUSH
361: N I F I=0:0 S I=$O(ZIPPED(I)) Q:'I  S SIZE=SIZE+$ZL(ZIPPED(I))
```

**Comment by shabiel** [comment]
@jensli @ksbhaskar did ask me to help fix this for you sooner rather than later, and it's now fixed, and pushed to master. It's fixed in [f301402](https://github.com/shabiel/M-Web-Server/commit/f3014023e14a828101acc139c83aab38b4cccb82).

FYI: We will be soon forking the M-Web-Server into a GT.M/YottaDB only version, which will be hosted on Gitlab. I will let you know when that happens.

**Comment by jensli** [comment]
Thanks!
